bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/VersioningTestHandler.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (C) 2011-2017 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.core.bind.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import org.commonjava.indy.bind.jaxrs.IndyResources;
+import org.commonjava.indy.core.ctl.StatsController;
+import org.commonjava.indy.stats.IndyVersioning;
+import org.commonjava.indy.util.ApplicationContent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatOkResponseWithJsonEntity;
+
+/**
+ * We do the following steps when we release a new apiVersion, e.g. 2
+ *
+ *  1) Update '<apiVersion>1 -> 2</apiVersion>' in indy-parent pom.xml. It will auto-update apiVersion in
+ *  'indy-version.properties' in indy-model-core-java during build. This apiVersion will be released along with the
+ *  client in one of the dependency jars.
+ *
+ *  2) If an api is to be changed in terms of req/resp message formats or the way how it handles the requests
```

Comment:
This process isn't exactly correct anymore. We're going to be using a path-matching file that will divert old request versions to some deprecated path, using a ServletFilter. In fact, all of the application/indy+json stuff needs to be removed, since it doesn't appear to work properly.